File: src/0345-Reverse-Vowels-of-a-String/0345.py
class Solution:
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        vowels = set(['a', 'u', 'e', 'i', 'o', 'A', 'U', 'E', 'I', 'O'])
        p = [i for i in s if i in vowels]
        return ''.join([i if i not in vowels else p.pop() for i in s])

if __name__ == "__main__":
    s = "hello"
    print(Solution().reverseVowels(s))

File: src/0345-Reverse-Vowels-of-a-String/0345.cpp
#include <string>
#include <iostream>
using namespace std;
static int x = []() {std::ios::sync_with_stdio(false); cin.tie(0); return 0; }();

class Solution
{
public:
    string reverseVowels(string s) 
    {
        string vowels = "aeiouAEIOU";
        int l = 0;
        int r = s.size() - 1;
        while (l < r)
        {
            l = s.find_first_of(vowels, l);
            r = s.find_last_of(vowels, r);
            if (l < r)
            {
                char tmp = s[l];
                s[l++] = s[r];
                s[r--] = tmp;
            }
        }
        return s;
    }
};
int main()
{
    string s = "hello";
    cout << Solution().reverseVowels(s) << endl;
    return 0;
}